src/pymor/algorithms/timestepping.py

```diff
@@ -10,221 +10,602 @@
+        M * d_t u + A(u, mu, t) = F(mu, t),
+                     u(mu, t_0) = u_0(mu),
+
+    by an explicit Euler time integration, implemented in :class:`ExplicitEulerIterator`.
 
     Parameters
     ----------
     nt
         The number of time-steps the time-stepper will perform.
+    num_values
+        The number of returned vectors of the solution trajectory. If `None`, each intermediate vector that is
+        calculated is returned. Else an interpolation of the calculated vectors on an equidistant temporal grid is
+        returned, using an appropriate interpolation of the respective time stepper.
     """
 
-    def __init__(self, nt):
+    IteratorType = ExplicitEulerIterator
+
+    def __init__(self, nt, num_values=None, interpolation='P1'):
+        super().__init__(num_values, interpolation)
+
+        assert isinstance(nt, Number)
+        assert nt > 0
+
         self.__auto_init(locals())
 
-    def solve(self, initial_time, end_time, initial_data, operator, rhs=None, mass=None, mu=None, num_values=None):
-        if mass is not None:
-            raise NotImplementedError
-        return explicit_euler(operator, rhs, initial_data, initial_time, end_time, self.nt, mu, num_values)
+
+class ExplicitRungeKuttaIterator(SingleStepTimeStepperIterator):
+
+    def __init__(self, stepper, t0, t1, U0, A, F, M, mu, iter, return_times):
+        super().__init__(stepper, t0, t1, U0, A, F, M, mu, iter, return_times)
+        self.__auto_init(locals())
+        self.dt = dt = (t1 - t0) / stepper.nt
+        # use the ones from base, these are checked and converted in super().__init__()
+        A, F, M, mu = self.A, self.F, self.M, self.mu
+
+        # prepare the function f in d_t y = f(t, y)
+        if not isinstance(M, IdentityOperator) and (not M.parametric or 't' not in M.parameters):
+            M = M.assemble(mu)
+        if not A.parametric or 't' not in A.parameters:
+            A = A.assemble(mu)
+
+        if isinstance(F, ZeroOperator):
+            if isinstance(M, IdentityOperator):
+                def f(t, y):
+                    mu_t = mu.with_(t=t)
+                    return -1*A.apply(y, mu=mu_t)
+            else:
+                def f(t, y):
+                    mu_t = mu.with_(t=t)
+                    return M.apply_inverse(-1*A.apply(y, mu=mu_t), mu=mu_t, initial_guess=y)
+        else:
+            if not F.parametric or 't' not in F.parameters:
+                F = F.assemble(mu)
+            if isinstance(M, IdentityOperator):
+                def f(t, y):
+                    mu_t = mu.with_(t=t)
+                    return F.as_vector(mu=mu_t) - A.apply(y, mu=mu_t)
+            else:
+                def f(t, y):
+                    mu_t = mu.with_(t=t) or Mu({'t': t})
+                    return M.apply_inverse(F.as_vector(mu=mu_t) - A.apply(y, mu=mu_t), mu=mu_t, initial_guess=y)
+
+        self.f = f
+
+    def _step_function(self, U_n, t_n):
+        c, A, b = self.stepper.butcher_array
```

Comment:
I would say that tradition requires calling this `butcher_tableau`?

Reply:
At least Wikipedia is in your favor, no objections...

Reply:
done in 09931564f26a3e5e921d4591e26332337c1cb58a